hooks/useGas.ts

```diff
@@ -0,0 +1,43 @@
+import { useMemo, useCallback } from 'react';
+import { useRecoilValue } from 'recoil';
+import { BigNumber } from 'ethers';
+
+import useSynthetixQueries from '@synthetixio/queries';
+import { customGasPriceState, gasSpeedState } from 'store/wallet';
+import { normalizeGasLimit, gasPriceInWei } from 'utils/network';
+
+const useGas = () => {
```

Comment:
this feels like something that should be in js-monorepo/queries

Reply:
seconding. thoughts?